src/main/java/uk/ac/ebi/embl/template/TemplateEntryProcessor.java

```diff
@@ -378,4 +395,122 @@ private void validateMarker(TemplateProcessorResultSet templateProcessorResultSe
+            SampleInfo sampleInfo=getSampleInfo(sample);
+            SampleEntity sampleEntity = getSampleEntity(sample);
+            updateSourceFeature(sourceFeature,sampleEntity,sampleInfo);
+            entry.addXRef(new XRef("BioSample", sample.getBioSampleId()));
+        }
+    }
+
+    /**
+     * This method works on ORGANISM_NAME field of the TSV file.
+     *  1) If the field value is taxid pattern then DO NOTHING and return.
+     *  2) If a sample is retrieved using the field value then return the sample.
+     *  
+     * @param templateVariables
+     * @param templateProcessorResultSet
+     * @param options
+     * @return
+     * @throws Exception
+     */
+    public Sample validateAndGetSample(TemplateVariables templateVariables, TemplateProcessorResultSet templateProcessorResultSet, SubmissionOptions options) throws Exception {
+
+        Map<String, String> tsvFieldMap = templateVariables.getVariables();
+        Sample sample = null;
+        boolean isDevmode = options.isDevMode;
+
+        // Iterate TSV header fields.
+        for (String tsvHeader : tsvFieldMap.keySet()) {
+            if (tsvHeader.equalsIgnoreCase(TemplateProcessorConstants.ORGANISM_TOKEN)) {
+                // When TSV header cell value is ORGANISM_NAME
+
+                if (matchesTaxId(tsvFieldMap.get(tsvHeader))) {
+                    // When field value is taxId pattern DO NOTHING
+                    break;
+                }
+
+                /** When tsv value do NOT match taxId pattern then retrieve sample assuming that the passed 
+                 *  value is sampleId / sampleAlias / bioSampleValue.
+                 *  If no sample is returned then the value is organism name.
+                 */
+                String sampleValue = tsvFieldMap.get(tsvHeader);
+                try {
+                    // Get sample from cache if exists. 
+                    sample=sampleCache.get(sampleValue);
+                    if(sample==null) {
+                        // Get sample using server API.
+                        CompleteSampleService completeSampleService = getCompleteSampleService(options.authToken.get(), isDevmode);
+                        sample = completeSampleService.getCompleteSample(sampleValue);
+                        sampleCache.put(sampleValue,sample);
+                    }
+                    break;
+                } catch (Exception e) {
+                    // On Exception return only INFO message so that execution continues as the given input is organism_name 
+                    ValidationMessage<Origin> message = new ValidationMessage<Origin>(Severity.INFO, "SampleSupportedCheck", sampleValue);
+                    templateProcessorResultSet.getValidationResult().append(new ValidationResult().append(message));
+                }
+            }
+        }
+        return sample;
+    }
+
+    private boolean matchesTaxId(String taxId){
```

Comment:
- if we have this method we might use the same name as what is being called isValidTaxId